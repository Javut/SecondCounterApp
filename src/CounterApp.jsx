import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({value}) => {

 const [counter, setCounter] = useState(value);

 const sumar = () => {
    setCounter(counter + 1);
 }

 const restar = () => {
    setCounter(counter - 1);
 }

 const reset = () => {
    setCounter(value);
 }

  return (
    <div>
        <h1>CounterApp</h1>
        <h2>{counter}</h2>
        
        {/* Las funciones al momento de llamarlas se deben pasar como una arrow function */}
        <button onClick={sumar}>+1</button>
        <button onClick={restar}>-1</button>
        <button aria-label="btn-reset" onClick={reset}>Reset</button>


    </div>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number
}