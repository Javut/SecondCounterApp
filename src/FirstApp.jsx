export const App = ( {title} ) => {

    return (
        <>
            <h1>Hola Mundo a Todos</h1>
            <h2 data-testid="test-title">{title}</h2>
        </>
    )
}