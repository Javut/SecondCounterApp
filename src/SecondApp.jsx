import PropTypes from "prop-types";



// const saludar = (name) => {
//     const saludo = `Hola ${name} como estas? `;
//     return saludo;
//  }

//props.title --> Al desestructurarse quedaria: "title"
export const SecondApp = ({title,subTitle}) => {

//  const newMessage = {
//     message: 'Hola Mundo',
//     title: 'Jaider'
//  };

// if (!title) { Es otra forma de valir su viene o no una prop peo no es recomendable al estar dentro de mi componente 
//   throw new Error('El title no existe');
// }



  return (
    <>
        {/* <h1>{saludar("messi")}</h1> */}
        <h2>{title}</h2>
        {/* <h2>{subtitulo}</h2> */}
        {/* <h1>{newMessage.message}</h1> */}
        {/* <h3>{JSON.stringify(newMessage)}</h3> */}
        <p>Este es un subtitulo</p>
    </>
  )
}

//Las PropTypes me sirven para definir el tipo a las properties
SecondApp.propTypes = { // De esta forma se definen las PropTypes de mi Aplicativo
  title: PropTypes.string.isRequired //Esto obliga a que se deba de enviar la property de title
}

//De esta forma se prodian definir properties por default 
SecondApp.defaultProps = { // En el hilo de ejecuci√≥n los defaultProps irian primero que las Proptypes 
  title: 'No hay titulo'
}