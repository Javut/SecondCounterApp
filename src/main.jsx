import React from 'react'
import ReactDom from 'react-dom/client'
import { CounterApp } from './CounterApp';
import { App } from './FirstApp'
import { SecondApp } from './SecondApp'
import './styles.css';




ReactDom.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={20}/>
        {/* <App title="Hola, Soy superman"/> */}
        {/* <SecondApp title="Mindsait"/>   */}
        {/* //De esta forma se le estarian mandando properties al componente Hijo <SecondApp></SecondApp> */}
    </React.StrictMode>
)


