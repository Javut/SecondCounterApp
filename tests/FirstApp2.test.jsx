import { render, screen } from "@testing-library/react"
import { App } from "../src/FirstApp"

describe('Pruebas en <FirstApp/>', () => {
  
    const title = 'Hola, Soy superman';
  
    test('should to do match with the snapshot ', () => {

        const {container} = render(<App title={title}/>);
        expect( container ).toMatchSnapshot();
      
    });

    test('debe de mostrar el mensaje "Hola, Soy superman" ', () => {
      
        render(<App title={title}/>);
        expect( screen.getByText(title)).toBeTruthy();

    });

    test('debe de mostrar el titulo en un h1', () => {
        
        render(<App title={title}/>);
        expect( screen.getByRole('heading', {level: 2} ).innerHTML).toContain(title);

    })
    
    
    

    

})
