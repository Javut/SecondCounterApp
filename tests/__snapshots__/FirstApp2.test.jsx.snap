// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> should to do match with the snapshot  1`] = `
<div>
  <h1>
    Hola Mundo a Todos
  </h1>
  <h2
    data-testid="test-title"
  >
    Hola, Soy superman
  </h2>
</div>
`;
